### PR TITLE
fix shellcheck error in tests/e2e/local/minikube/setup_host.sh

### PR DESCRIPTION
**Pull request by stewart-yu**
In PR #8668, ignored the `vm-driver` error, so fix it

### CODE REVIEW

File: tests/e2e/local/minikube/setup_host.sh [hunk 1]
```diff
@@ -14,12 +14,13 @@ case "${OSTYPE}" in
...
 
-# shellcheck disable=SC2153
-if [ ! -z "$VM_DRIVER" ]; then
-  vm_driver="$VM_DRIVER"
-fi
+# VM for Minikube, the default is decided by your os type if not specified
+# e.g.:
+#   hyperkit default to darwin os
+#   kvm2 default to Debian or Ubuntu os
+VM_DRIVER=${VM_DRIVER:-$(vm_driver)}
```
Comment (m1093782566) [inline tests/e2e/local/minikube/setup_host.sh]: Why  can not be:

```
vm_driver=${vm_driver:-$(VM_DRIVER)}
```

?

File: tests/e2e/local/minikube/setup_host.sh [hunk 2]
```diff
@@ -14,12 +14,13 @@ case "${OSTYPE}" in
   *) echo "unsupported: ${OSTYPE}" ;;
 esac
 
-# shellcheck disable=SC2153
-if [ ! -z "$VM_DRIVER" ]; then
-  vm_driver="$VM_DRIVER"
-fi
+# VM for Minikube, the default is decided by your os type if not specified
```
Comment (m1093782566) [inline tests/e2e/local/minikube/setup_host.sh]: What about change the comment to:

Virtual machine driver for minikube, the default value is decided by your OS type if it's not specified...

File: tests/e2e/local/minikube/setup_host.sh (hunk 1 above)
Comment (stewart-yu) [reply tests/e2e/local/minikube/setup_host.sh]: If change like that, i will add `VM_DRIVER=${VM_DRIVER:-""}` , will check later

Comment (m1093782566) [reply tests/e2e/local/minikube/setup_host.sh]: Seems you are right.

I am fine with the change if it's necessary for fixing the shell check errors.

File: tests/e2e/local/minikube/setup_host.sh [hunk 3]
```diff
@@ -14,12 +14,14 @@ case "${OSTYPE}" in
...
 
-# shellcheck disable=SC2153
-if [ ! -z "$VM_DRIVER" ]; then
-  vm_driver="$VM_DRIVER"
-fi
+# Virtual machine driver, the default value is decided by your OS type if it's not specified,
+# e.g.:
+#   hyperkit default to darwin os
+#   kvm2 default to Debian or Ubuntu os
+# Besides, you can set any vm-driver you like via exporting `VM_DRIVER`for you environment.
```
Comment (m1093782566) [inline tests/e2e/local/minikube/setup_host.sh]: typo: your -> your

Comment (m1093782566) [reply tests/e2e/local/minikube/setup_host.sh]: nit: should put in a white space between words `VM_DRIVER` and for.

**Comment by stewart-yu** [comment]
just do simple test, works:
no export `VM_DRIVER` in my environment(`ubuntu`), run `tests/e2e/local/minikube/setup_host.sh`, part of output like this:
```
Using kvm2 as VM for Minikube.
tests/e2e/local/minikube/setup_host.sh: line 27: minikube: command not found
Starting Minikube.
```
after export `export VM_DRIVER=test-vm` in environment(`ubuntu`),  run `tests/e2e/local/minikube/setup_host.sh`, part of output like this:
```
Using test-vm as VM for Minikube.
tests/e2e/local/minikube/setup_host.sh: line 27: minikube: command not found
Starting Minikube.
sudo: minikube: command not found
```
 The error for that not install `minikube`, but we can see that the `VM_DRIVER` take effect

**Comment by m1093782566** [comment]
@stewart-yu 

Seems you failed to install minikube. Did you check your network and install minikube manually?

**Comment by m1093782566** [comment]
/lgtm

/assign @hklai 

for final review and approval.

File: tests/e2e/local/minikube/setup_host.sh (hunk 3 above)
Comment (stewart-yu) [reply tests/e2e/local/minikube/setup_host.sh]: fixed

**Comment by stewart-yu** [comment]
not intall `minikube` in my local, so the error are normally. Thanks for your review @m1093782566 :upside_down_face:

**Comment by stewart-yu** [comment]
prow/istio-unit-tests.sh

**Comment by m1093782566** [comment]
Let me try to approve this PR.

/approve

**Comment by stewart-yu** [comment]
may be the robot are `unhappy`:wink:

**Comment by m1093782566** [comment]
@stewart-yu Can you try to assign this PR to me?

**Comment by stewart-yu** [comment]
The robot response so stranger
/cc @hzxuzhonghu

**Comment by m1093782566** [comment]
Probably I need to join istio-hacker team.

@rshriram,, am I right?

**Comment by m1093782566** [comment]
Seems I can make the ci-robot happy now :)

@stewart-yu Thanks for catching this!

**Pull request merged**